sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/query/AbstractReactiveCosmosQuery.java

```diff
@@ -61,12 +61,11 @@ public Object execute(Object[] parameters) {
      * @param returnedType The return type of the method
      * @return the execution type needed to handle the query
      */
-    protected ReactiveCosmosQueryExecution getExecution(ReturnedType returnedType) {
+    protected ReactiveCosmosQueryExecution getExecution(ReactiveCosmosParameterAccessor accessor, ReturnedType returnedType) {
         if (isDeleteQuery()) {
             return new ReactiveCosmosQueryExecution.DeleteExecution(operations);
         } else if (isPageQuery()) {
-            throw new IllegalArgumentException("Paged Query is not supported by reactive cosmos "
-                + "db");
+            return new ReactiveCosmosQueryExecution.PagedExecution(operations, accessor.getPageable());
```

Comment:
`isPageQuery()` checks if the return type is of `Page.java` type, which in this case it is not. Since `CosmosPagedFlux` is not type of `Page.java`. How will it go in this else if condition?